### ISSUE TITLE
Keycloak.init is not called again when the initConfig prop changes

### ISSUE THREAD

**Issue by VincentvD** [panz3r/react-keycloak#28]
Keycloak allows you to pass in an initial token and refreshToken on init. However the `componentDidUpdate` in the KeycloakProvider only checks whether `keycloak` has changed. 

Ok, so the steps to reproduce are not easily reproduced but I'll try and describe my situation. 
- Anonymous user opens application
- Keycloak is initialized (without tokens in initConfig)
- User registers in app and the API response contains tokens.
- I update the initConfig with the tokens that is passed along as a prop to KeycloakProvider
- No update occurs.

The expected behaviour is that init will fire after initConfig has been changed. I have tested this locally with this solution which seems to work.

```
componentDidUpdate({ initConfig: prevInitConfig, keycloak: prevKeycloak }) {
    if (
      this.props.keycloak !== prevKeycloak ||
      this.props.initConfig !== prevInitConfig
    ) {
      // De-init previous Keycloak instance
      prevKeycloak.onReady = null
      prevKeycloak.onAuthSuccess = null
      prevKeycloak.onAuthError = null
      prevKeycloak.onAuthRefreshSuccess = null
      prevKeycloak.onAuthRefreshError = null
      prevKeycloak.onAuthLogout = null
      prevKeycloak.onTokenExpired = null

      // Reset state
      this.setState({ ...initialState })
      // Init new Keycloak instance
      this.init()
    }
  }
```

**Comment by panz3r** [comment]
Hi @VincentvD ,

Thanks for your feedback.

I just released a new version of `react-keycloak` ([`6.1.1`](https://github.com/panz3r/react-keycloak/releases/tag/v6.1.1)) which includes the fix for this issue.

Just for reference I implemented a slightly modified version of your proposed solution because when dealing with `Objects` it is always better to do a `deep compare` (using functions like [`_.isEqual`](https://lodash.com/docs/4.17.15#isEqual) or [`deep-compare`](https://github.com/dangvanthanh/deep-compare) )  instead of a `shallow compare` (using `===`).

**Comment by VincentvD** [comment]
Thanks for the quickfix. Indeed my proposed solution was too shallow...
